### ISSUE TITLE
3 wishes for djinn?

### ISSUE THREAD

**Issue by yrmvgh** [jeremygurr/dcssca#391]
No scroll use is a pretty serious "stasis" esque conduct which is fun and interesting to work around, but it also makes weapon-using djinn pretty dependent on what they find in the dungeon - can't enchant, can't acquire.

Could also go the undead route and allow scroll use when in forms (you're not on fire at that pont). That's less interesting, though.

**Comment by yrmvgh** [comment]
On the topic of djinn, how about we allow Glow to increase your LOS for now, even though that's "bad", and we can start identifying _how_ bad it is, exactly, and what sort of benefits are possible.

I know you want to implement a tricksy non-reciprocal LOS at some point but it'd be interesting to try a normal LOS expansion for now.

**Comment by jeremygurr** [comment]
allowing the djinn to use scrolls while transformed greatly weakens the limitation, making it almost meaningless, and somewhat tedious to get around. That limitation is important to the character of the djinni and is also important to balancing out their other way OP attributes. 

I already tried the increasing LOS thing and it doesn't work too well. I have a better solution now though I have on my list to implement. Every character will have their "LOS", but also a maximum targetting range and a maximum visibility range, 3 different distances. The LOS will function like before, where monsters can't see you if they aren't in your LOS. And maximum targetting range is never allowed to be larger the LOS. However, it is possible for a player to see beyond their "LOS", which I know makes the acronym a misnomer. So the glow 3 will allow players to see one tile further than their LOS, but nothing will allow them to target into that range. This means players will be able to see monsters before the monsters can see them, which is a pretty big advantage. 

This also allows us to implement a "blindness" nerf, which doesn't affect LOS, but reduces maximum targetting distance. You can see monsters outside of your vision range, but only after they hit you first, as if they were invisible. However, to keep it from being too difficult, the game indicates what monsters are in your LOS, even if you can't see them, in the sidebar, and the messages still report that a certain monster / unique has entered your LOS, even if you can't see exactly where they are, enabling you to take emergency measures if they are too dangerous.

**Comment by jeremygurr** [comment]
Ok fine. Djinni should start with 3 wish scrolls. I can't deny that the Djinni screams for this flavor to be added. Djinni are so powerful already, but maybe I can take away fire healing or something. 

So a wish scroll of course is inferior to the new and improved acquirement scroll (which I haven't finished yet, but will allow you to select from 10 known items, instead of ending up with lames ones 2/3 of the time). 

But they are still pretty cool. A wish scroll allows you to pick a base item class, could be amulet, ring, short blade, long blade, axe, polearm, staff, bow, crossbow, sling, blowgun, or mace. 

Then the game randomly generates which specific base item you will get, and adds a bunch of random attributes, equal chance of good or bad. 

So you may ask for a long blade, and get a falcion -3 [+rF -rC *Confuse], just as wishes often backfire for the player. But you could also get something nice. And since you have 3 wishes, the odds are decent you will get something usable.

**Comment by SleepySteel** [comment]
It sounds nice for flavor, but I don't think it should come at the cost of genuinely unique traits that allow new strategies- fire healing is a really cool part of djinni characters because it lets you do wacky stuff like point blank living flame that other characters generally can't manage.

**Comment by jeremygurr** [comment]
I agree about not wanting to lose unique traits, but the trait you are referring to wouldn't be lost in this proposal. If Dj went from healing in fire to just being immune to fire (as they were originally), they would still be able to use inner flame and immolation safely at point blank range. Not to mention self-fireballing or firestorming.

**Comment by jeremygurr** [comment]
But it does take away the ability of a Dj to cast conjure flame and then stand in the flame to heal. This is kind of cheap anyway, so probably should go, but it was fun while it lasted.

**Comment by G0rg0** [comment]
I don't think it's cheap at all - they effectively burn HP to cast spells and it seems pretty decently balanced. The healing in flames only reduces HP loss, it doesn't come close to counteracting it, and the clouds don't last long when you are sitting in one. 

If the clouds lasted as long as usual and the healing was enough that you could just blast away then I would say tweaking was in order. I say leave it as is!

Also - scrolls of acquirement are known as scrolls of disappointment for a reason. Yeah, they'll give you something nice on occasion, but most often just junk.

**Comment by SleepySteel** [comment]
I also don't think it's too cheap- the main problem seems to be that djinn doesn't "suck up" the clouds hard enough if there's one at all.

**Comment by Sandman25DCSS** [comment]
Please don't add the wishes to nightmare mode.
